src/client/browser/idle-page/index.js

```diff
@@ -3,36 +3,46 @@ import StatusIndicator from './status-indicator';
 import COMMAND from '../../../browser/connection/command';
 
 
-const CHECK_STATUS_DELAY = 1000;
-
 const createXHR = () => new XMLHttpRequest();
 
+const CHECK_STATUS_DELAY = 1000;
 
 class IdlePage {
-    constructor (statusUrl, heartbeatUrl, initScriptUrl, options = {}) {
+    constructor (statusUrl, heartbeatUrl, initScriptUrl, { retryTestPages } = {}) {
         this.statusUrl       = statusUrl;
+        this.heartbeatUrl    = heartbeatUrl;
+        this.initScriptUrl   = initScriptUrl;
         this.statusIndicator = new StatusIndicator();
 
-        if (options.retryTestPages)
-            browser.enableRetryingTestPages();
+        this.retryTestPages = retryTestPages;
 
-        browser.startHeartbeat(heartbeatUrl, createXHR);
-        browser.startInitScriptExecution(initScriptUrl, createXHR);
+        document.title = '[' + document.location.toString() + ']';
+    }
 
-        this._checkStatus();
+    async _pollStatus () {
+        let { command } = await browser.checkStatus(this.statusUrl, createXHR);
 
-        document.title = '[' + document.location.toString() + ']';
+        while (command.cmd === COMMAND.idle) {
+            await browser.delay(CHECK_STATUS_DELAY);
+
+            ({ command } = await browser.checkStatus(this.statusUrl, createXHR));
```

Comment:
`await browser.checkStatus(this.statusUrl, createXHR);`

Reply:
Do you suggest to remove additional brackets? They are required if we need to destruct objects into already declared variable.

Reply:
Sorry, it's my mistake. All is OK.